### PR TITLE
flood: init at 4.2.0

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

Second try of this PR: https://github.com/NixOS/nixpkgs/pull/102552

Using shared node packages now, and also bumping the package to its latest version.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104735` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>lumo</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>bat-extras.prettybat</li>
    <li>iosevka</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>ycmd</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104735` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>rambox-pro</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>iosevka</li>
    <li>lumo</li>
  </ul>
</details>
<details>
  <summary>39 packages built:</summary>
  <ul>
    <li>antora</li>
    <li>bat-extras.prettybat</li>
    <li>bitwarden-cli</li>
    <li>castnow</li>
    <li>create-cycle-app</li>
    <li>emojione</li>
    <li>epgstation</li>
    <li>etcher</li>
    <li>flood</li>
    <li>image_optim</li>
    <li>imgbrd-grabber</li>
    <li>joplin</li>
    <li>mastodon-bot</li>
    <li>mirakurun</li>
    <li>netlify-cli</li>
    <li>parity-ui</li>
    <li>pm2</li>
    <li>redoc-cli</li>
    <li>ripcord</li>
    <li>slack</li>
    <li>teleprompter</li>
    <li>thelounge</li>
    <li>twemoji-color-font</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-html</li>
    <li>vimPlugins.coc-java</li>
    <li>vimPlugins.coc-lists</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-prettier</li>
    <li>vimPlugins.coc-smartf</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vimPlugins.coc-yaml</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
    <li>wasm-text-gen</li>
    <li>yaml-language-server</li>
    <li>ycmd</li>
  </ul>
</details>

File: pkgs/applications/networking/flood/default.nix [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{ pkgs, nodejs, nodePackages, stdenv, lib, ... }:
```
Comment (midchildan) [inline pkgs/applications/networking/flood/default.nix:1]: I believe this should be moved to `pkgs/development/node-packages/default.nix`.

**Review by midchildan — CHANGES_REQUESTED** [review]

**Review by midchildan — APPROVED** [review]

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 104735` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>dat</li>
  </ul>
</details>
<details>
  <summary>41 packages built:</summary>
  <ul>
    <li>antora</li>
    <li>bat-extras.prettybat</li>
    <li>bitwarden-cli</li>
    <li>castnow</li>
    <li>create-cycle-app</li>
    <li>emojione</li>
    <li>epgstation</li>
    <li>etcher</li>
    <li>flood</li>
    <li>image_optim</li>
    <li>imgbrd-grabber</li>
    <li>iosevka</li>
    <li>joplin</li>
    <li>lumo</li>
    <li>mastodon-bot</li>
    <li>mirakurun</li>
    <li>netlify-cli</li>
    <li>parity-ui</li>
    <li>pm2</li>
    <li>redoc-cli</li>
    <li>ripcord</li>
    <li>slack (slack-dark)</li>
    <li>teleprompter</li>
    <li>thelounge</li>
    <li>twemoji-color-font</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-html</li>
    <li>vimPlugins.coc-java</li>
    <li>vimPlugins.coc-lists</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-prettier</li>
    <li>vimPlugins.coc-smartf</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vimPlugins.coc-yaml</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
    <li>wasm-text-gen</li>
    <li>yaml-language-server</li>
    <li>ycmd</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@thiagokokada Can you solve the merge conflict?

**Comment by thiagokokada** [comment]
@SuperSandro2000 Yeah, I am working on it. Sadly running `generate.sh` script (on my machine at least) is pretty slow...

**Comment by thiagokokada** [comment]
@SuperSandro2000 @midchildan rebased again.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 Yeah, I am working on it. Sadly running `generate.sh` script (on my machine at least) is pretty slow...

It does a lot of network requests. I can also run it and merge it in a few hours after it.

File: pkgs/development/node-packages/node-env.nix [hunk 2]
```diff
@@ -516,7 +516,7 @@ let
     stdenv.mkDerivation {
       name = "node-shell-${name}-${version}";
 
-      buildInputs = [ python nodejs ] ++ stdenv.lib.optional (stdenv.isLinux) util-linux ++ buildInputs;
+      buildInputs = [ python nodejs ] ++ stdenv.lib.optional (stdenv.isLinux) utillinux ++ buildInputs;
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/node-env.nix:519]: Why is the script doing this? utillinux is an alias which fails the pipeline. I fixed that and amended it to get this through before we have another merge conflict.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104735` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>iosevka</li>
    <li>lumo</li>
  </ul>
</details>
<details>
  <summary>29 packages built:</summary>
  <ul>
    <li>antora</li>
    <li>castnow</li>
    <li>emojione</li>
    <li>epgstation</li>
    <li>fast-cli</li>
    <li>flood</li>
    <li>gtop</li>
    <li>image_optim</li>
    <li>joplin</li>
    <li>mirakurun</li>
    <li>netlify-cli</li>
    <li>pm2</li>
    <li>postcss-cli</li>
    <li>redoc-cli</li>
    <li>ripcord</li>
    <li>twemoji-color-font</li>
    <li>vimPlugins.coc-lists</li>
    <li>vimPlugins.coc-markdownlint</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-prettier</li>
    <li>vimPlugins.coc-rust-analyzer</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-tabnine</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vimPlugins.coc-yaml</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>vscode-extensions.ms-python.vscode-pylance</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
    <li>yaml-language-server</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104735` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>lumo</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>iosevka</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
  </ul>
</details>

**Pull request merged**